src/main/java/ryze/Ryze.java

```diff
@@ -0,0 +1,380 @@
+                printDivider();
+            }
+            else {
+                System.out.println("Invalid task number");
+            }
+        } catch (NumberFormatException e) {
+            System.out.println("Invalid task number format.");
+        }
+    }
+
+    private static void addDeadlineTask(String line) throws InvalidNumberArguments, IOException {
+        String[] parts = line.split("/by ", 2);
+        if (parts.length == 2){
+        String description = parts[0].replace(DEADLINE_COMMAND, "").trim();
+        String deadline = parts[1].trim();
+        Task newDeadline = new Deadline(description, deadline);
+        listOfChatHistory.add(newDeadline);
+        echo(newDeadline.toString(), listOfChatHistory.size());
+        appendToFile(newDeadline.toData());
+        }
+        else {
+            throw new InvalidNumberArguments("Invalid command format for adding deadline.");
+        }
+    }
+
+    private static void addEventTask(String line) throws InvalidNumberArguments, IOException {
+        String[] parts = line.split("/from | /to ", 3);
+        if (parts.length == 3) {
+            String eventDescription = parts[0].replace(EVENT_COMMAND, "").trim();
+            String startTime = parts[1].trim();
+            String endTime = parts[2].trim();
+            Task newEvent = new Event(eventDescription, startTime, endTime);
+            listOfChatHistory.add(newEvent);
+            echo(newEvent.toString(), listOfChatHistory.size());
+            appendToFile(newEvent.toData());
+        } else {
+            throw new InvalidNumberArguments("Invalid command format for adding event.");
+        }
+    }
+
+    private static void markOrUnmarkTask(String line, String command) {
+        try {
+            int taskNumber = Integer.parseInt(line.split(" ")[1]);
+            if (taskNumber > 0 && taskNumber <= listOfChatHistory.size()) {
+                Task task = listOfChatHistory.get(taskNumber -1);
+                if (command.equals(MARK_COMMAND)) {
+                    if (task.isDone()) {
+                        System.out.println("You have already completed this task.");
+                    } else {
+                        task.markAsDone();
+                        printDivider();
+                        System.out.println("Nice! I've marked this task as done:");
+                        System.out.printf("  %s%n%n", task);
+                    }
+                } else {
+                    if (!task.isDone()) {
+                        System.out.println("You have already uncompleted this task.");
+                    } else {
+                        task.markAsNotDone();
+                        printDivider();
+                        System.out.println("Okay! I've marked this task as not done:");
+                        System.out.printf("  %s%n%n", task);
+                    }
+                }
+                printDivider();
+            } else {
+                printDivider();
```

Comment:
Try to avoid deep nesting, you can consider using if else if where appropriate.
E.g.
```
int subsidy() {
    int subsidy;
    if (!age) {
        if (!sub) {
            if (!notFullTime) {
                subsidy = 500;
            } else {
                subsidy = 250;
            }
        } else {
            subsidy = 250;
        }
    } else {
        subsidy = -1;
    }
    return subsidy;
}
```
versus
```
int calculateSubsidy() {
    int subsidy;
    if (isSenior) {
        subsidy = REJECT_SENIOR;
    } else if (isAlreadySubsidized) {
        subsidy = SUBSIDIZED_SUBSIDY;
    } else if (isPartTime) {
        subsidy = FULLTIME_SUBSIDY * RATIO;
    } else {
        subsidy = FULLTIME_SUBSIDY;
    }
    return subsidy;
}

```